windows/hid.c

```diff
@@ -183,19 +188,44 @@ static int lookup_functions()
 #endif /* HIDAPI_USE_DDK */
 
 struct hid_device_ {
-		HANDLE device_handle;
-		BOOL blocking;
-		USHORT output_report_length;
-		unsigned char *write_buf;
-		size_t input_report_length;
-		USHORT feature_report_length;
-		unsigned char *feature_buf;
-		wchar_t *last_error_str;
-		BOOL read_pending;
-		char *read_buf;
-		OVERLAPPED ol;
-		OVERLAPPED write_ol;
-		struct hid_device_info* device_info;
+	HANDLE device_handle;
+	BOOL blocking;
+	USHORT output_report_length;
+	unsigned char *write_buf;
+	size_t input_report_length;
+	USHORT feature_report_length;
+	unsigned char *feature_buf;
+	wchar_t *last_error_str;
+	BOOL read_pending;
+	char *read_buf;
+	OVERLAPPED ol;
+	OVERLAPPED write_ol;
+	struct hid_device_info* device_info;
+};
+
+static struct hid_hotplug_context {
+	/* Win32 notification handle */
+	HCMNOTIFICATION notify_handle;
+
+	/* Critical section (faster mutex substitute), for both cached device list and callback list changes */
+	CRITICAL_SECTION critical_section;
+
+	int critical_section_ready;
```

Comment:
I think this should be a BOOL type.

Reply:
Given that it's declared as `typedef int BOOL;` I don't see much difference :)

Reply:
The difference in discipline and code consistency. It does matter a lot.
```suggestion
	BOOL critical_section_ready;
```
and `BOOL` is way more explicit/readable